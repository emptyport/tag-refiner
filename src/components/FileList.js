import React, { Component } from 'react';
import { Line } from 'rc-progress';

class FileList extends React.Component {
  constructor(props) {
    super(props);
    console.log(props);
    this.state = {
      messageTrace: '',
      processingList: []
    };

    

  }

  render() {
    
    let fileList = this.props.files.map(function(entry) {
      return (
        <li className="file-entry" key={entry.id}>
          <div className="file-name">
            {entry.file}
          </div>
          <Line className="file-progress" percent={entry.progress} />
          <div className="file-status">
            {entry.status}
          </div>
        </li>
      );
    });

    return (
      <div>
        <ul>
          {fileList}
        </ul>
      </div>
    );
  }
}

export default FileList;
