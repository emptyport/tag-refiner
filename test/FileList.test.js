import React from 'react';
import FileList from '../src/components/FileList';
import renderer from 'react-test-renderer';

test('FileList initializes correctly', () => {
  const component = renderer.create(
    <FileList files={[]} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.type).toBe('div');
  expect(tree.children.length).toBe(1);
  expect(tree.children[0].type).toBe('ul');
});

test('FileList displays files', () => {
  let mzMLFile = 'dummy/path/file.mzML';
  let fileStatus = 'Pending';
  let fileList = [
    {
      id: 0,
      progress: 0,
      file: mzMLFile,
      status: fileStatus
    }
  ];
  const component = renderer.create(
    <FileList files={fileList} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.type).toBe('div');
  expect(tree.children.length).toBe(1);
  expect(tree.children[0].type).toBe('ul');

  let child = tree.children[0];
  expect(child.children.length).toBe(1);

  let fileItem = child.children[0];
  expect(fileItem.type).toBe('li');
  expect(fileItem.children[0].type).toBe('div');
  expect(fileItem.children[1].type).toBe('svg');
  expect(fileItem.children[2].type).toBe('div');

  let filename = fileItem.children[0].children[0];
  expect(filename).toBe(mzMLFile);

  let status = fileItem.children[2].children[0];
  expect(status).toBe(fileStatus);
});