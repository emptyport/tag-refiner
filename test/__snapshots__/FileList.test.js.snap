// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList displays files 1`] = `
<div>
  <ul>
    <li
      className="file-entry"
    >
      <div
        className="file-name"
      >
        dummy/path/file.mzML
      </div>
      <svg
        className="rc-progress-line file-progress"
        preserveAspectRatio="none"
        style={Object {}}
        viewBox="0 0 100 4"
      >
        <path
          className="rc-progress-line-trail"
          d="M 2,2
           L 98,2"
          fillOpacity="0"
          stroke="#D9D9D9"
          strokeLinecap="round"
          strokeWidth={1}
        />
        <path
          className="rc-progress-line-path"
          d="M 2,2
           L 98,2"
          fillOpacity="0"
          stroke="#2db7f5"
          strokeLinecap="round"
          strokeWidth={4}
          style={
            Object {
              "strokeDasharray": "100px, 100px",
              "strokeDashoffset": "100px",
              "transition": "stroke-dashoffset 0.3s ease 0s, stroke 0.3s linear",
            }
          }
        />
      </svg>
      <div
        className="file-status"
      >
        Pending
      </div>
    </li>
  </ul>
</div>
`;

exports[`FileList initializes correctly 1`] = `
<div>
  <ul />
</div>
`;
